scheduler/core/components/collector/__init__.py

```diff
@@ -481,3 +482,31 @@ def night_configurations(self,
                 self.get_night_events(site).time_grid[night_idx].datetime.date() - Collector._DAY
             )
             for night_idx in night_indices]
+
+    def time_accounting(self, site_plans: Plans) -> None:
+        """ Do time accounting for Plans in both site but for only one night.
+            As implemented right now it does not consider any interruptions.
+        """
+        for plan in site_plans:
+            for v in plan.visits:
+                # Update Observation from Collector.
+                observation = self.get_observation(v.obs_id)
+                # check that Observation is Observed
+                observation.status = ObservationStatus.ONGOING
+
+                # Update by atom in the sequence
+                for atom_idx, atom in enumerate(observation.sequence):
+
+                    atom.program_used = atom.prog_time
+                    atom.partner_used = atom.part_time
+
+                    # Charge acquisition to the first atom
+                    if atom_idx == 0:
+                        if observation.obs_class == ObservationClass.PARTNERCAL:
+                            atom.program_used += observation.acq_overhead
+                        elif (observation.obs_class == ObservationClass.SCIENCE or
+                              observation.obs_class == ObservationClass.PROGCAL):
+                            atom.program_used += observation.acq_overhead
```

Comment:
Instead of doing this, you should just be able to move this out of the for loop and do it directly for `first_atom = observation.sequence[0]`.

Reply:
The problem is the for has to be changed as well because it would replace the values in the first iteration and if the first atom is skip entirely then now we have to add the first atom.progtime as well. I guess is just matter of personal preference?

Reply:
if atom_idx == 0 should be if atom_idx = v.atom_start_idx.